pyomo/contrib/viewer/pyomo_viewer.py

```diff
@@ -12,9 +12,29 @@
 Run a Pyomo model viewer with embeded Jupyter QtConsole
 """
 
+import sys
 import logging
+import traceback
 _log = logging.getLogger(__name__)
 from pyomo.contrib.viewer.ui import *
+import pyomo.environ as pyo
+
+def log_unhandled_exception(etype, evalue, etrace):
+    """
+    This will log an exception and show a message box about what went wrong, but
+    let the application keep running.  It could be frustrating if an unhandled
+    exception caused the program to immediatly terminate.
+    """
+    try:
+        _log.critical("unhandled exception", exc_info=(etype, evalue, etrace))
+        msgBox = QMessageBox()
+        msgBox.setWindowTitle("Error")
+        msgBox.setText("Unhandled Exception")
```

Comment:
Should there also be a disclaimer that the error should be reported to the developers and that the GUI should be exited immediately?  After an unhandled exception like this, I am not sure there is much we can say about the validity / stability of the underlying object model, right?

Reply:
You're right.  I'll make that update as soon as I get a chance.

Reply:
I added to the message,. While I was at it I added handling of the KeyboardInterrupt exception and bailed on the ui.py import if pyqt isn't available.   I have some dummy classes so I could test the data model, but it was causing the ui module import to get too far before failing in a weird way, so this make thing a bit easier to understand.